src/utils/editor/source-editor.js

```diff
@@ -18,40 +20,57 @@ require("codemirror/addon/search/searchcursor");
 // while shifting to a line which was initially out of view.
 const MAX_VERTICAL_OFFSET = 3;
 
+type SourceEditorOpts = {
+  enableCodeFolding: boolean,
+  extraKeys: Object,
+  gutters: string[],
+  lineNumbers: boolean,
+  lineWrapping: boolean,
+  matchBrackets: boolean,
+  mode: string,
+  readOnly: boolean,
+  showAnnotationRuler: boolean,
+  theme: string,
+  value: string,
+};
+
 class SourceEditor {
-  constructor(opts) {
+  opts: SourceEditorOpts;
+  editor: any;
+
+  constructor(opts: SourceEditorOpts) {
     this.opts = opts;
   }
 
-  appendToLocalElement(node) {
+  appendToLocalElement(node: any) {
     this.editor = CodeMirror(node, this.opts);
   }
 
   destroy() {
     // No need to do anything.
   }
 
-  get codeMirror() {
+  get codeMirror(): any {
     return this.editor;
   }
 
-  setText(str) {
+  setText(str: string) {
     this.editor.setValue(str);
   }
 
   getText() {
     return this.editor.getValue();
   }
 
-  setMode(value) {
+  setMode(value: Object) {
```

Comment:
Heh. I think sometimes it's a string. Let's hunt this down. I believe we keep a modded dict in utils source